src/extension/mapml/src/main/java/org/geoserver/mapml/gwc/gridset/MapMLGridsets.java

```diff
@@ -281,4 +329,127 @@ public String getIdentifier() {
+                        .collect(Collectors.toList());
+
+        Collections.sort(filteredNames);
+        return filteredNames;
+    }
+
+    private static boolean canBeSupportedAsTiledCRS(GridSet gridSet) {
+        String name = gridSet.getName();
+        if (FIXED_NAMES.contains(name) || name.contains(":")) {
+            return false;
+        }
+        GridSetLevelType levelType = MapMLGridsets.getLevelType(getLevelNamesFromGridSet(gridSet));
+        return levelType.isNumeric() || levelType.isPrefixed();
+    }
+
+    private static List<String> getLevelNamesFromGridSet(GridSet gridSet) {
+        List<String> levelNames = new ArrayList<>();
+        for (int i = 0; i < gridSet.getNumLevels(); i++) {
+            Grid grid = gridSet.getGrid(i);
+            levelNames.add(grid.getName());
+        }
+
+        return levelNames;
+    }
+
+    /**
+     * Extracts the level names from a list of {@link TileMatrix} objects.
+     *
+     * @param tileMatrices the list of {@link TileMatrix} objects to process; must not be {@code
+     *     null}.
+     * @return a list of level names (identifiers) as {@link String} objects, one for each {@link
+     *     TileMatrix} in the input list. Returns an empty list if {@code tileMatrices} is empty.
+     * @throws NullPointerException if {@code tileMatrices} or any of its elements are {@code null}.
+     */
+    public static List<String> getLevelNamesFromTileMatrixList(List<TileMatrix> tileMatrices) {
+        List<String> levelNames = new ArrayList<>();
+
+        // Iterate over each TileMatrix and add its identifier to the list
+        for (TileMatrix tileMatrix : tileMatrices) {
+            levelNames.add(tileMatrix.getIdentifier().toString());
+        }
+
+        return levelNames;
+    }
+
+    /**
+     * Determines the type of levels from a given list of level strings.
+     *
+     * <p>If the levels are all numeric, it returns a {@link GridSetLevelType} with the {@code
+     * numeric} flag set to {@code true}. If a common prefix exists, it sets the {@code prefixed}
+     * flag to {@code true} and assigns the common prefix to the {@code prefix} field.
+     *
+     * @param levels a list of level identifiers to process; can be {@code null} or empty.
+     * @return a {@link GridSetLevelType} object indicating the level type:
+     *     <ul>
+     *       <li>{@code numeric = true} if all levels are numeric.
+     *       <li>{@code prefixed = true} with a {@code prefix} value if levels share a common
+     *           prefix.
+     *     </ul>
+     *     Returns {@code null} if the input list is {@code null} or empty.
+     */
+    public static GridSetLevelType getLevelType(List<String> levels) {
+        if (levels == null || levels.isEmpty()) {
+            return null;
+        }
+        // First check: levels are simple numbers:
+        GridSetLevelType levelType = new GridSetLevelType();
+        String numericPattern = "-?\\d+(\\.\\d+)?";
```

Comment:
Ouch, this way the pattern is re-compiled over and over. Create a Pattern as a static variable instead.